pkg/account/deployer/client.go

```diff
@@ -145,54 +146,89 @@ func (d *accountManagementClient) upsertPolicy(ctx context.Context, policyLevel
 func (d *accountManagementClient) upsertGroup(ctx context.Context, groupId string, group Group) (remoteId, error) {
 	if groupId != "" {
 		logr.FromContextOrDiscard(ctx).V(1).Info("Trying to update group with origin object ID (UUID) " + groupId)
-		resp, err := d.client.GroupManagementAPI.EditGroup(ctx, d.accountInfo.AccountUUID, groupId).PutGroupDto(group).Execute()
-		defer closeResponseBody(resp)
-
-		if err = d.handleClientResponseError(resp, err, "unable to update group with UUID: "+groupId); err != nil {
+		existingGroup, err := d.getGroupByID(ctx, groupId)
+		if err != nil {
 			return "", err
 		}
-		return groupId, nil
+
+		return d.updateExistingGroup(ctx, *existingGroup, group)
 	}
 
+	existingGroupsWithName, err := d.getGroupsByName(ctx, group.Name)
+	if err != nil {
+		return "", err
+	}
+
+	if len(existingGroupsWithName) == 0 {
+		return d.createGroup(ctx, group)
+	}
+
+	if len(existingGroupsWithName) > 1 { // shouldn't happen
+		logr.FromContextOrDiscard(ctx).V(-1).Info("Updating multiple policies with name " + group.Name + ". Updating group with UUID " + existingGroupsWithName[0].GetUuid())
```

Comment:
not your change, but IIRC, negative verbosity is discarded/there's no Warning level in the logger combination we're using. (Bernd, did we every look into this any further?) So may as well log this on just Info

Reply:
I'm too new to know, but is there a reason we log directly with `logr` here? Rather than use say `internal/log/log.go`?

Reply:
I can only guess, but I'd assume this happened with the idea that this client might move to the core library. 
(@warber @jskelin can one of you recall for sure why we use logr here?)

Reply:
yes, the idea was to also move the "smart" AM client to the lib repository, which obviously never happened.

Reply:
if we see any value/potential in it being reused, we should just move it IMO. If we don't, as @arthurpitman says, we might as welluse our logger directly and actually get warning level logs.